platform/website/src/components/auth/auth-dialog.svelte

```diff
@@ -48,56 +50,38 @@
 	let turnstileToken = "";
 	let loggingIn = false;
 
-	let username = newField({
-		id: "username",
-		type: "text",
-		label: "Username",
-		autoComplete: "username",
-		extra: {
-			timeout: undefined as NodeJS.Timeout | undefined,
-		},
-		reload() {
-			username.touched = username.value.length > 0;
+	$: $authDialog, resetAllFields();
 
+	let usernameValue: string;
+	let usernameStatus: FieldStatus;
+	let usernameValidationTimeout: number | NodeJS.Timeout;
+	function usernameValidate(v: string): Promise<FieldStatus> {
+		return new Promise((resolve) => {
+			globalMessage = "";
 			if ($authDialog === AuthDialog.Login) {
-				username.message = "";
-				username.status = "";
-			}
-
-			if (username.touched) {
-				username.update(username.value);
-			}
-		},
-		update(value) {
-			username.value = value;
-
-			// We clear the timeout so that we dont send a request to the server on every character change.
-			clearTimeout(username.extra.timeout);
-
-			// When we are in login mode, we dont need to validate the username.
-			if ($authDialog !== AuthDialog.Register) {
+				resolve({ type: FieldStatusType.Success });
 				return;
 			}
 
+			clearTimeout(usernameValidationTimeout);
```

Comment:
Does this cause a memory leak?
What happens when you drop this if the promise has never resolved?

Reply:
I think this is solved by the refactoring. Is it? I'm not sure